spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/health/DatastoreHealthIndicator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.health;
+
+import java.util.UUID;
+
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.cloud.gcp.data.datastore.core.DatastoreTemplate;
+import org.springframework.cloud.gcp.data.datastore.repository.support.SimpleDatastoreRepository;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+/**
+ * Simple implementation of a {@link HealthIndicator} returning status information for Google cloud
+ * data store.
+ *
+ * @author Raghavan N S
+ * @author Srinivasa Meenavalli
+ */
+@Component
+public class DatastoreHealthIndicator extends AbstractHealthIndicator {
+
+	private static final Status DATASTORE_HEALTH = new Status("DATASTORE_HEALTH");
+
+	private final SimpleDatastoreRepository<HealthCheckEntity, String> simpleDatastoreRepository;
```

Comment:
I think it's too much machinery to use a repository to do a simple liveliness check.
Perhaps querying some metadata or statistics like `__namespace__` or `__Stat_Total__` directly using the `Datastore` class would be more efficient.
You can even just query for a non-existing kind:

```java
@Autowired
Datastore datastore;
...
datastore.run(Query.newKeyQueryBuilder().setKind("RandomKindName").build());
```
If no exception is thrown, Datastore must be up.

Reply:
I agree. I will use statistics query